tests/jointcalTestBase.py

```diff
@@ -0,0 +1,311 @@
+# See COPYRIGHT file at the top of the source tree.
+
+from __future__ import division, absolute_import, print_function
+
+import numpy as np
+import collections
+import os
+
+from lsst.meas.astrom import LoadAstrometryNetObjectsTask, LoadAstrometryNetObjectsConfig
+from lsst.afw import geom, table
+
+from lsst.jointcal import jointcal
+
+arcsec_per_radian = 206265.
+
+
+class JointcalTestBase(object):
+    """
+    Base class for jointcal tests, to genericize some test running and setup.
+
+    Derive from this first, then from TestCase.
+    """
+
+    def setUp(self):
+        self.doPlot = False  # don't make plots unless specifically requested
+        self.matchRadius = 0.1*geom.arcseconds  # match sources within 0.1" for RMS statistics
+
+    def tearDown(self):
+        if getattr(self, 'jointcalTask', None) is not None:
+            del self.jointcalTask
+        if getattr(self, 'jointcalTask', None) is not None:
+            del self.catalogs
+
+    def _prep_reference_loader(self, center, radius):
+        """
+        Setup an astrometry.net reference loader.
+
+        @param center (afw.coord) The center of the field you're testing on.
+        @param radius (afw.geom.angle) The radius to load objects around center.
+        """
+        refLoader = LoadAstrometryNetObjectsTask(LoadAstrometryNetObjectsConfig())
+        # Make a copy of the reference catalog for in-memory contiguity.
+        self.reference = refLoader.loadSkyCircle(center, radius, filterName='r').refCat.copy()
```

Comment:
Actually, this brings up a good point: I'm assuming we have all the "del thing"s in our test tearDowns specifically for MemoryTestCase and the behavior of C++ objects in python? Because otherwise, I don't see how they're needed, if the object will just get rebuilt in setUp.